File: QtModel/Core/QtServer.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace QtModel.Core;

public static class QtServer
{
    private static readonly HttpClient Client = new();

    public static string Url { get; set; } = "http://localhost:55125/pythonForQt/";
    public static string MergeString { get; set; } = string.Empty;
    public static bool QtMerge { get; set; }
    public static string QtVersion { get; set; } = "1.2.3";

    public static string? SendCommand(string command = "", string header = "")
        => SendCommandAsync(command, header).GetAwaiter().GetResult();

    public static async Task<string?> SendCommandAsync(string command = "", string header = "")
    {
        using var request = new HttpRequestMessage(HttpMethod.Post, Url)
        {
            Content = new StringContent(command, Encoding.UTF8)
        };
        if (!string.IsNullOrEmpty(header))
        {
            request.Content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue(header);
        }

        var response = await Client.SendAsync(request).ConfigureAwait(false);
        var text = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
        return response.StatusCode switch
        {
            System.Net.HttpStatusCode.OK => text,
            System.Net.HttpStatusCode.BadRequest => throw new InvalidOperationException(text),
            System.Net.HttpStatusCode.RequestEntityTooLarge =>
                throw new InvalidOperationException("请求体过大，请拆分请求或调整服务端限制"),
            System.Net.HttpStatusCode.GatewayTimeout =>
                throw new InvalidOperationException("服务端或反向代理超时，请增加最大等待时间"),
            _ => throw new InvalidOperationException("连接错误，请重新尝试")
        };
    }

    public static Task<string?> SendDictAsync(string header, object? payload = null)
    {
        if (payload is null)
        {
            return SendCommandAsync(string.Empty, header);
        }

        var dictionary = JsonSerializer.Serialize(payload, new JsonSerializerOptions
        {
            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
            Encoder = System.Text.Encodings.Web.JavaScriptEncoder.UnsafeRelaxedJsonEscaping
        });

        return SendCommandAsync(dictionary, header);
    }

    public static string? SendDict(string header, object? payload = null)
        => SendDictAsync(header, payload).GetAwaiter().GetResult();
}

File: qtmodel/core/qt_server.py
import json
import threading
from typing import Optional
import requests

class QtServer:
    URL: str = "http://localhost:55125/pythonForQt/"
    QT_VERSION: str = "2.2.3"
    _session = requests.Session()
    _lock = threading.Lock()

    @staticmethod
    def send_command(command: str = "", header: str = ""):
        try:
            with QtServer._lock:
                response = QtServer._session.post(
                    QtServer.URL,
                    headers={'Content-Type': f'{header}'},
                    data=command.encode('utf-8'),
                    timeout=(3, 60)
                )

            if response.status_code == 200:
                return response.text
            elif response.status_code == 400:
                raise Exception(response.text)
            elif response.status_code == 413:
                raise Exception("请求体过大，请拆分请求或调整服务端请求限制")
            elif response.status_code == 504:
                raise Exception("服务端处理超时，请增加最大等待时间")
            else:
                raise Exception(f"连接错误，请重新尝试。HTTP {response.status_code}: {response.text}")
        except requests.exceptions.RequestException as ex:
            raise Exception(f"请求失败: {ex}") from ex

    @staticmethod
    def send_dict(header: str, payload: Optional[dict] = None):
        try:
            if not payload:
                return QtServer.send_command(header=header, command="")
            payload = dict(payload)
            if "version" not in payload:
                payload["version"] = QtServer.QT_VERSION
            json_string = json.dumps(payload, ensure_ascii=False)
            return QtServer.send_command(header=header, command=json_string)
        except Exception as ex:
            raise Exception(str(ex)) from ex
